File: sdks/rust/examples/basic_usage.rs
//! Basic usage example for Nexus Rust SDK

use nexus_sdk::*;
use std::collections::HashMap;

#[tokio::main]
async fn main() -> Result<()> {
    // Create a client
    let client = NexusClient::new("http://localhost:15474")?;

    // Check server health
    tracing::info!("Checking server health...");
    let healthy = client.health_check().await?;
    tracing::info!("Server is healthy: {}", healthy);

    // Get database statistics
    tracing::info!("\nGetting database statistics...");
    let stats = client.get_stats().await?;
    tracing::info!("Nodes: {}", stats.catalog.node_count);
    tracing::info!("Relationships: {}", stats.catalog.rel_count);
    tracing::info!("Labels: {}", stats.catalog.label_count);
    tracing::info!("Relationship types: {}", stats.catalog.rel_type_count);

    // Execute a simple Cypher query
    tracing::info!("\nExecuting Cypher query...");
    let result = client
        .execute_cypher("RETURN 'Hello, Nexus!' as greeting", None)
        .await?;
    tracing::info!("Query result: {:?}", result);

    // Create a label
    tracing::info!("\nCreating label...");
    let label_response = client.create_label("Person".to_string()).await?;
    tracing::info!("Label creation: {}", label_response.message);

    // List all labels
    tracing::info!("\nListing labels...");
    let labels = client.list_labels().await?;
    tracing::info!("Labels: {:?}", labels.labels);

    // Create a node
    tracing::info!("\nCreating node...");
    let mut properties = HashMap::new();
    properties.insert("name".to_string(), Value::String("Alice".to_string()));
    properties.insert("age".to_string(), Value::Int(30));
    properties.insert("active".to_string(), Value::Bool(true));

    let create_response = client
        .create_node(vec!["Person".to_string()], properties)
        .await?;
    tracing::info!("Created node with ID: {}", create_response.node_id);

    // Get the node
    tracing::info!("\nGetting node...");
    let get_response = client.get_node(create_response.node_id).await?;
    if let Some(node) = get_response.node {
        tracing::info!("Node ID: {}", node.id);
        tracing::info!("Labels: {:?}", node.labels);
        tracing::info!("Properties: {:?}", node.properties);
    }

    // Update the node
    tracing::info!("\nUpdating node...");
    let mut update_properties = HashMap::new();
    update_properties.insert("age".to_string(), Value::Int(31));
    let update_response = client
        .update_node(create_response.node_id, update_properties)
        .await?;
    tracing::info!("Update result: {}", update_response.message);

    // Create a relationship type
    tracing::info!("\nCreating relationship type...");
    let rel_type_response = client.create_rel_type("KNOWS".to_string()).await?;
    tracing::info!("Relationship type creation: {}", rel_type_response.message);

    // List relationship types
    tracing::info!("\nListing relationship types...");
    let types = client.list_rel_types().await?;
    tracing::info!("Relationship types: {:?}", types.types);

    tracing::info!("\nExample completed successfully!");
    Ok(())
}

File: sdks/python/examples/basic_usage.py
"""Basic usage example for Nexus Python SDK."""

import asyncio
from nexus_sdk import NexusClient


async def main():
    """Run basic usage example."""
    # Create a client
    async with NexusClient("http://localhost:15474") as client:
        # Execute a Cypher query
        result = await client.execute_cypher("MATCH (n) RETURN n LIMIT 10", None)
        print(f"Found {len(result.rows)} rows")

        # Create a node
        create_response = await client.create_node(
            labels=["Person"], properties={"name": "Alice", "age": 30}
        )
        print(f"Created node with ID: {create_response.node_id}")

        # Get a node
        node = await client.get_node(create_response.node_id)
        if node:
            print(f"Node: {node}")

        # Update a node
        update_response = await client.update_node(
            create_response.node_id, properties={"age": 31}
        )
        print(f"Updated node: {update_response.node}")

        # Create another node
        create_response2 = await client.create_node(
            labels=["Person"], properties={"name": "Bob", "age": 25}
        )

        # Create a relationship
        rel_response = await client.create_relationship(
            source_id=create_response.node_id,
            target_id=create_response2.node_id,
            rel_type="KNOWS",
            properties={"since": 2020},
        )
        print(f"Created relationship with ID: {rel_response.relationship_id}")

        # Get database statistics
        stats = await client.get_stats()
        print(f"Database stats: {stats}")

        # Health check
        healthy = await client.health_check()
        print(f"Server is healthy: {healthy}")


if __name__ == "__main__":
    asyncio.run(main())

